### PR TITLE
More vuetify widgets

### PR DESCRIPTION
**Pull request by M-A-Boucher**
Add a few useful Vuetify widgets.

### CODE REVIEW

File: src/Themes/HackathonTheme/Views/Widget-VAlert.liquid [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+{% assign icon = Model.ContentItem.Content.VAlert.Icon.Text %}
+{% assign border = Model.ContentItem.Content.VAlert.Border.Text %}
+
+<v-alert {{ Model.ContentItem.Content.VAlert.Props.Values | join: ' ' | raw }}
+  {% if color %}color="{{ color }}"{% endif %}
+  {% if type %}type="{{ type }}"{% endif %}
+  {% if elevation %}elevation="{{ elevation }}"{% endif %}
+  {% if icon %}icon="{{ icon }}"{% endif %}
+  {% if border %}border="{{ border }}"{% endif %}>
+  {{ Model.ContentItem.Content.HtmlBodyPart.Html | sanitize_html | raw }}
```
Comment (jptissot) [inline src/Themes/HackathonTheme/Views/Widget-VAlert.liquid:13]: Here, I suggest you simply render the HtmlBodyPart.
`{{ Model.Content.HtmlBodyPart | shape_render }}`

File: src/Themes/HackathonTheme/Views/Widget-VCard.liquid [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+<v-card>
+  {% if Model.ContentItem.Content.VCard.Title.Text %}
+    <v-card-title>{{ Model.ContentItem.Content.VCard.Title.Text }}</v-card-title>
+  {% endif %}
+  {% if Model.ContentItem.Content.VCard.Subtitle.Text %}
+    <v-card-subtitle>{{ Model.ContentItem.Content.VCard.Subtitle.Text }}</v-card-subtitle>
+  {% endif %}
+  {% if Model.ContentItem.Content.VCard.Content.Html != "" %}
+    <v-card-text>
+      {{ Model.ContentItem.Content.VCard.Content.Html | sanitize_html | shortcode | raw }}
```
Comment (jptissot) [inline src/Themes/HackathonTheme/Views/Widget-VCard.liquid:12]: Render the shape here as well

File: src/Themes/HackathonTheme/Views/Widget-VInput.liquid [hunk 3]
```diff
@@ -0,0 +1,31 @@
+{% assign model = Model.ContentItem.Content.VInput.Model.Text %}
```
Comment (jptissot) [inline src/Themes/HackathonTheme/Views/Widget-VInput.liquid]: Should this be part of the VueForms module ?

File: src/Themes/HackathonTheme/Views/Widget-VTextField.liquid [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{% assign model = Model.ContentItem.Content.VTextField.Model.Text %}
```
Comment (jptissot) [inline src/Themes/HackathonTheme/Views/Widget-VTextField.liquid:1]: Same, should this be in the VueForms module ?

File: src/Themes/HackathonTheme/Migrations.cs [hunk 5]
```diff
@@ -270,5 +278,741 @@ private void VContainer()
...
+
+        private void VCard()
+        {
+            _contentDefinitionManager.AlterTypeDefinition("VCard", type => type
+                .DisplayedAs("VCard")
+                .Stereotype("Widget")
+                .WithPart("VCard", part => part
+                    .WithPosition("1")
+                )
+                .WithPart("TitlePart", part => part
```
Comment (jptissot) [inline src/Themes/HackathonTheme/Migrations.cs:425]: For these, try using the extension method WithTitlePart() I added to the extensions

File: src/Themes/HackathonTheme/Migrations.cs [hunk 6]
```diff
@@ -270,5 +278,741 @@ private void VContainer()
...
+
+        private void VExpansionPanel()
+        {
+            _contentDefinitionManager.AlterTypeDefinition("VExpansionPanel", type => type
+                .DisplayedAs("VExpansionPanel")
+                .Stereotype("Widget")
+                .WithPart("VExpansionPanel", part => part
+                    .WithPosition("1")
+                )
+                .WithPart("TitlePart", part => part
```
Comment (jptissot) [inline src/Themes/HackathonTheme/Migrations.cs:471]: WitTitlePart() extension method

**Review by jptissot — CHANGES_REQUESTED** [review]

File: src/Themes/HackathonTheme/Views/Widget-VInput.liquid (hunk 3 above)
Comment (M-A-Boucher) [reply src/Themes/HackathonTheme/Views/Widget-VInput.liquid]: Probably makes more sense in the VueForms module, for sure.

File: src/Themes/HackathonTheme/Views/Widget-VCard.liquid (hunk 2 above)
Comment (M-A-Boucher) [reply src/Themes/HackathonTheme/Views/Widget-VCard.liquid:12]: Will it render the shortcodes correctly?

**Review by jptissot — APPROVED** [review]

**Pull request merged**